server/src/main/java/org/opensearch/indices/recovery/RecoveryTarget.java

```diff
@@ -141,18 +103,6 @@ public RecoveryTarget retryCopy() {
         return new RecoveryTarget(indexShard, sourceNode, listener);
     }
 
-    public ActionListener<Void> markRequestReceivedAndCreateListener(long requestSeqNo, ActionListener<Void> listener) {
-        return requestTracker.markReceivedAndCreateListener(requestSeqNo, listener);
-    }
-
-    public long recoveryId() {
-        return recoveryId;
-    }
-
-    public ShardId shardId() {
```

Comment:
Not particularly biased, but supporting `public ShardId shardId()`  as a shortcut to `indexShard().getId()`  could be handy (too many `recoveryTarget.indexShard().getId()` calls)

Reply:
+1, I think we should add this to ReplicationTarget.